src/common/checksums.cpp

```diff
@@ -360,8 +360,16 @@ void ValidateChecksumHeader::slotChecksumCalculated(const QByteArray &checksumTy
         emit validationFailed(tr("The checksum header contained an unknown checksum type '%1'").arg(QString::fromLatin1(_expectedChecksumType)));
         return;
     }
-    if (checksum != _expectedChecksum) {
```

Comment:
Hm it would be sufficient to check:
```c++
// allow leading padding
if (_expectedChecksum.endsWith(checksum)) {
```


https://doc.qt.io/qt-5/qbytearray.html#endsWith

Reply:
I think the padding prefix character should be specified, because otherwise the probability of false positive matches will be increased as far as I can see.

I'd suggest `_expectedChecksum.split('0', QString::SkipEmptyParts).join("") == checksum)`.